src/sage/games/hexad.py

```diff
@@ -545,77 +543,77 @@ def find_hexad(self, pts):
         MINIMOG = self.minimog
         L = set(pts)
         LL = L.copy()
-        pts_at_infty = set([MINIMOG[0][2],MINIMOG[2][1],MINIMOG[0][0]])
+        pts_at_infty = {MINIMOG[0][2], MINIMOG[2][1], MINIMOG[0][0]}
         # recall & means intersection
         L2 = LL & pts_at_infty
         if len(L2) == 3:  # must be type 0 (line + pts at infty)
             H, WHAT = self.find_hexad0(LL - pts_at_infty)
             return H, WHAT
         if len(L2) == 2:  # type 0 or 3
             if (MINIMOG[0][2] in LL and MINIMOG[2][1] in LL):
-                H, WHAT = self.find_hexad3(LL - set([MINIMOG[0][2],MINIMOG[2][1]]),
+                H, WHAT = self.find_hexad3(LL - {MINIMOG[0][2], MINIMOG[2][1]},
                                            MINIMOG[0][2], MINIMOG[2][1])
-                if H != []:   # must be type 3
+                if H:   # must be type 3
                     return list(H), WHAT
                 if H == []:   # could be type 0
```

Comment:
This could be `if not H:`, or we can remove this test as we checked `if H:` 2 lines before.

The same situation appears multiple times below.